http_server/session.pony

```diff
@@ -243,5 +244,10 @@ trait tag Session
     """
     None
 
+  be upgrade(notify: TCPConnectionNotify iso) =>
+    """
+    Upgrade this TCP connection to another handler
+    """
```

Comment:
I know that "upgrade" is [the official HTTP jargon for this](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Upgrade), but it might be confusing to people at first. I recommend something like "upgrade_protocol" to make this more clear. I also recommend linking to those docs, as below:

```suggestion
  be upgrade_protocol(notify: TCPConnectionNotify iso) =>
    """
    Upgrade this TCP connection to another handler.
    See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Upgrade
    """
```